### ISSUE TITLE
coeio threads may use destroyed cord data 

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/tarantool#1491]
```
==1840== Thread 10 memcheck-amd/eio:
==1840== Invalid read of size 4
==1840==    at 0x644C0C: evpipe_write (ev.c:2310)
==1840==    by 0x64C053: ev_async_send (ev.c:4809)
==1840==    by 0x4CE10E: coeio_want_poll_cb (coeio.c:102)
==1840==    by 0x50B9FA: etp_proc (etp.c:369)
==1840==    by 0x62E7453: start_thread (pthread_create.c:334)
==1840==    by 0x70A6EEC: clone (in /lib/x86_64-linux-gnu/libc-2.22.so)
==1840==  Address 0x7cc9048 is 616 bytes inside a block of size 1,064 free'd
==1840==    at 0x4C2AE6B: free (vg_replace_malloc.c:530)
==1840==    by 0x643A9E: ev_realloc_emul (ev.c:1587)
==1840==    by 0x643AE2: ev_realloc (ev.c:1602)
==1840==    by 0x64726E: ev_loop_destroy (ev.c:2917)
==1840==    by 0x4F8312: cord_destroy (fiber.c:676)
==1840==    by 0x4F8C91: cord_join (fiber.c:781)
==1840==    by 0x45C4B9: wal_writer_stop() (wal.cc:330)
==1840==    by 0x44A7FC: box_free (box.cc:1183)
==1840==    by 0x42ABDE: tarantool_free() (main.cc:469)
==1840==    by 0x6FF3E07: __run_exit_handlers (in /lib/x86_64-linux-gnu/libc-2.22.so)
==1840==    by 0x6FF3E54: exit (in /lib/x86_64-linux-gnu/libc-2.22.so)
==1840==    by 0x6FDE616: (below main) (in /lib/x86_64-linux-gnu/libc-2.22.so)
==1840==  Block was alloc'd at
==1840==    at 0x4C29C0F: malloc (vg_replace_malloc.c:299)
==1840==    by 0x4C2BE6F: realloc (vg_replace_malloc.c:785)
==1840==    by 0x643A90: ev_realloc_emul (ev.c:1585)
==1840==    by 0x643AE2: ev_realloc (ev.c:1602)
==1840==    by 0x64734E: ev_loop_new (ev.c:2966)
==1840==    by 0x4F87C5: cord_start (fiber.c:744)
==1840==    by 0x4F907E: cord_costart (fiber.c:931)
==1840==    by 0x45C390: wal_writer_start(wal_mode, char const*, tt_uuid const*, vclock*, long) (wal.cc:297)
==1840==    by 0x44B00B: box_init() (box.cc:1406)
==1840==    by 0x44B0BD: box_load_cfg (box.cc:1432)
==1840==    by 0x42ABAA: load_cfg (main.cc:454)
==1840==    by 0x45ED7E: lbox_cfg_load(lua_State*) (cfg.cc:60)
```

**Comment by rtsisyk** [comment]
one more case:
```
/lib/x86_64-linux-gnu/libc.so.6(+0x7811e)[0x7f9f20c0711e]
tarantool parallel.test.lua <running>[0x4c7911]
tarantool parallel.test.lua <running>[0x49c890]
tarantool parallel.test.lua <running>[0x49d29f]
tarantool parallel.test.lua <running>(phia_env_delete+0x14c)[0x49cfdc]
tarantool parallel.test.lua <running>(_ZN10PhiaEngineD1Ev+0x40)[0x4960a0]
tarantool parallel.test.lua <running>(_ZN10PhiaEngineD0Ev+0x19)[0x496199]
tarantool parallel.test.lua <running>(_Z15engine_shutdownv+0xce)[0x448c2e]
tarantool parallel.test.lua <running>(box_free+0x58)[0x458538]
tarantool parallel.test.lua <running>(_Z14tarantool_freev+0x24)[0x42dca4]
/lib/x86_64-linux-gnu/libc.so.6(+0x35e08)[0x7f9f20bc4e08]
/lib/x86_64-linux-gnu/libc.so.6(+0x35e55)[0x7f9f20bc4e55]
tarantool parallel.test.lua <running>[0x63c23c]
tarantool parallel.test.lua <running>[0x5d615b]
tarantool parallel.test.lua <running>(lua_pcall+0x1d8)[0x5a3638]
tarantool parallel.test.lua <running>(lbox_call+0x23)[0x5209e3]
tarantool parallel.test.lua <running>[0x519b98]
```

**Comment by mejedi** [comment]
Fixed in 916f1b8d58
